server/tailsql/options.go

```diff
@@ -610,3 +627,42 @@ func DefaultCheckQuery(q Query) (Query, error) {
 	}
 	return q, nil
 }
+
+// Queryable is the interface used to issue SQL queries to a database.
+type Queryable interface {
+	// Query issues the specified SQL query in a transaction and returns the
+	// matching result set, if any.
+	Query(ctx context.Context, sql string) (RowSet, error)
```

Comment:
nit: QueryContext to match `database/sql`? and if you add params you could probably just pass in a `*sql.DB`

Reply:
Good idea, re: params (done).

Re: `*sql.DB`, not quite, because its QueryContext doesn't return a RowSet. 

As we talked about, I looked into doing this with a generic, but then the type parameters infect everything that touches the type (which is kind of a lot). The problem is we want multiple different types of Queryable (e.g., to support cfgdb), which means it leaks into DBSpec, and then it leaks into dbHandle and dbUpdate and Options—it's kind of a mess. 

So I think the wrapper is preferable, since we really do want a dynamic dispatch anyway.

Reply:
Ah right, sg